cmd/cli/root.go

```diff
@@ -30,99 +21,44 @@ import (
 	"github.com/akuity/kargo/internal/cli/cmd/update"
 	"github.com/akuity/kargo/internal/cli/cmd/version"
 	clicfg "github.com/akuity/kargo/internal/cli/config"
-	"github.com/akuity/kargo/internal/cli/option"
+	cliutil "github.com/akuity/kargo/internal/cli/util"
 )
 
-// rootState holds state used internally by the root command.
-type rootState struct {
-	localServerListener net.Listener
-}
-
-func NewRootCommand(
-	cfg clicfg.CLIConfig,
-	opt *option.Option,
-	rs *rootState,
-) (*cobra.Command, error) {
+func NewRootCommand(cfg clicfg.CLIConfig) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:               "kargo",
 		DisableAutoGenTag: true,
 		SilenceUsage:      true,
-		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			ctx := buildRootContext(cmd.Context())
-
-			if opt.UseLocalServer {
-				restCfg, err := config.GetConfig()
-				if err != nil {
-					return errors.Wrap(err, "get REST config")
-				}
-				client, err :=
-					kubernetes.NewClient(ctx, restCfg, kubernetes.ClientOptions{})
-				if err != nil {
-					return errors.Wrap(err, "error creating Kubernetes client")
-				}
-				l, err := net.Listen("tcp", "127.0.0.1:0")
-				if err != nil {
-					return errors.Wrap(err, "start local server")
-				}
-				rs.localServerListener = l
-				srv := api.NewServer(
-					apiconfig.ServerConfig{
-						LocalMode: true,
-					},
-					client,
-					client,
-				)
-				go srv.Serve(ctx, l) // nolint: errcheck
-				opt.LocalServerAddress = fmt.Sprintf("http://%s", l.Addr())
-			}
-			return nil
-		},
```

Comment:
Not a blocker... more of a what if...

Knowing that much of this will shortly re-surface under a new `kargo server` command, what would you think about moving this code under a new, _hidden_ command and we'll work work on that as time permits and unhide it when it's ready?

Dealer's choice.

Reply:
This is actually a great suggestion to keep the code around, even if it would abort as soon as the command is run.

I will implement this with a TODO note with a rough outline of what the future should bring.